### PR TITLE
feat: add a helm chart for kommander-lifecycle

### PR DESCRIPTION
**Pull request by Azhovan**
**What problem does this PR solve?**:
implementing required HelmChart 

**Which issue(s) does this PR fix?**:
https://d2iq.atlassian.net/browse/D2IQ-95249

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

**Comment by gracedo** [comment]
I read the comment in the JIRA - will we consider moving this chart out of k-apps into kommander? This is just temporary first step?

**Comment by Azhovan** [comment]
> I read the comment in the JIRA - will we consider moving this chart out of k-apps into kommander? This is just temporary first step?

yes this is only for testing

**Comment by Azhovan** [comment]
@mikolajb @palexster @gracedo 

During my testing, I noticed that simply putting this new chart(defined by this PR) and its HelmRelease doesn't is not applied to clusters.

To dig dipper, I created a [new PR](https://github.com/mesosphere/kommander-applications/pull/915), this time, I uploaded the chart into a temporary repo in my github account [here](https://github.com/Azhovan/helm-charts). by doing so 
[new helmrepository](https://github.com/mesosphere/kommander-applications/pull/915/files#diff-83395c2103e5a96ed11549c024a3ad43e38a58da18d9f40407ea168e09418b14R3) showed up in the cluster
image

 However helm release was not created in cluster.

To make sure helmrelease definition is correct I applied it  manually to the cluster

image

and it succeeded, which to me it means flux only apply helmreleases defined within `services` directory

**Comment by gracedo** [comment]
@Azhovan I'm not sure if I follow completely, but just the changes in this PR will not automatically get the helmrelease installed to the cluster - you would need to create a flux kustomization that is pointing to this `kommander-lifecycle/` path in the gitrepo

File: kommander-lifecycle/chart/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v2
+name: kommander-lifecycle
```
Comment (mikolajb) [inline kommander-lifecycle/chart/Chart.yaml:2]: `kommander-lifecycle` or kommander-operator, wdyt?

cc: @palexster

File: kommander-lifecycle/chart/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+appVersion: "0.1.0"
+version: 0.1.0
+description: A controller that will be responsible for the lifecycle of the KommanderCore resource.
```
Comment (mikolajb) [inline kommander-lifecycle/chart/Chart.yaml:5]:
```suggestion
description: kommander-lifecycle helm chart.
```

**Review by mikolajb — COMMENTED** [review]
Consider removing `.helmignore`, `_helpers.tpl` and `values.yaml` (they are not used).

File: kommander-lifecycle/chart/Chart.yaml (hunk 2 above)
Comment (Azhovan) [reply kommander-lifecycle/chart/Chart.yaml]: thanks!

**Comment by palexster** [comment]
Do we need the templates for basic operator scaffolding in this PR or is this expected in another PR?

**Comment by mikolajb** [comment]
> Do we need the templates for basic operator scaffolding in this PR or is this expected in another PR?

@Azhovan I would suggest a separate PR, wdyt?

**Review by palexster — APPROVED** [review]

File: kommander-lifecycle/chart/Chart.yaml (hunk 1 above)
Comment (Azhovan) [reply kommander-lifecycle/chart/Chart.yaml]: done

**Review by mikolajb — APPROVED** [review]

**Pull request merged**